### PR TITLE
Add CSC/DT rechits to AOD

### PR DESCRIPTION
**Pull request by kakwok**
#### PR description:

This PR adds CSC/DT rechits to the AOD datatier, which are the essential inputs for searches of LLP decaying in the muon system.

Similar to #34066

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: RecoLocalMuon/Configuration/python/RecoLocalMuon_EventContent_cff.py [hunk 1]
```diff
@@ -5,9 +5,11 @@
 # AOD content
 RecoLocalMuonAOD = cms.PSet(
     outputCommands = cms.untracked.vstring(
+        'keep *_dt1DRecHits_*_*',
         'keep *_dt4DSegments_*_*', 
         'keep *_dt4DCosmicSegments_*_*',
-        'keep *_cscSegments_*_*', 
+        'keep *_cscSegments_*_*',
+        'keep *_csc2DRecHits_*_*',
```
Comment (slava77) [inline RecoLocalMuon/Configuration/python/RecoLocalMuon_EventContent_cff.py:12]: we discussed with PPD and concluded to restrict this to bParking in 120X as well.
@rappoccio

**Comment by slava77** [comment]
@kakwok 
this PR needs an update to use bParking
https://github.com/cms-sw/cmssw/pull/34072#discussion_r648764615

**Comment by slava77** [comment]
test parameters:
- workflows = 136.898,1304.182

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34072	7a1dce8
- code changes are in line with the PR description and the follow up review: CSC and DT rechits are added to the AOD datatier *only in bParking* setup
    - @kakwok please update the PR description to clarify this restriction
- jenkins tests pass and comparisons with the baseline show no differences
    -  the step3.root output of the prod-like bParking workflow 1304.182 is confirmed to have the new collections, as expected
    - the expected file size increase is about 1.5% as found in 10_6_X tests https://github.com/cms-sw/cmssw/pull/34066#issuecomment-858051096

The future proposed plan is to replace direct addition of hits with some reduced/slimmed information. The selection logic will be based on what is found in the bParking UL production.

**Comment by qliphy** [comment]
+1

**Pull request merged**